hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketSetPropertyRequest.java

```diff
@@ -289,6 +291,11 @@ public boolean checkQuotaBytesValid(OMMetadataManager metadataManager,
 
     if (quotaInBytes > OzoneConsts.QUOTA_RESET) {
       totalBucketQuota = quotaInBytes;
+      if (quotaInBytes < dbBucketInfo.getUsedBytes()) {
+        throw new OMException("Cannot update bucket quota. Requested " +
+            "spaceQuota less than used spaceQuota.",
+            OMException.ResultCodes.QUOTA_ERROR);
```

Comment:
We can use `QUOTA_EXCEEDED` instead of `QUOTA_ERROR`, which gives a better error code. Same for the other change as well.

Reply:
I have used QUOTA_ERROR, as its not correct value as already used quota. This is inline that volume quota not cleared while set as its not in correct state to use.

QUOTA_EXCEED is used when set value crosses quota as assigned by volume.

Reply:
Makes sense. I'm good with the changes.